adapter-kafka/pom.xml

```diff
@@ -76,6 +76,12 @@
             <artifactId>commons-configuration2</artifactId>
             <version>2.8.0</version>
         </dependency>
+        <dependency>
+            <groupId>io.nosqlbench</groupId>
+            <artifactId>adapter-pulsar</artifactId>
```

Comment:
please add a comment that explains why we need this dependency

Reply:
https://github.com/nosqlbench/nosqlbench/pull/1143#issuecomment-1450375869

Reply:
I didn't mean to add a comment on the PR but to also drop a line like:

```
    <!-- This dependency is needed in order to re-use some utilities from the Pulsar adapter code -->
```

Reply:
Oh ok fixed.

Reply:
@jshook @yabinmeng - I didn't give enough context about this dep. 

I was suggesting to create a new module as a shared util module (generating metrics tracking message duplicates, message loss and re-ordered messages) as this could be easily reused by some module other than `adapter-pulsar` and `adapter-kafka`. For example a module which might need that in the future is `adapter-s4j`.

Reply:
I decoupled moved the reused code in `adapters-api` module in `io.nosqlbench.engine.api.metrics.ReceivedMessageSequenceTracker`. Also see: https://github.com/nosqlbench/nosqlbench/pull/1143/commits/b33389824980c41c7d227a465ffa5875287763d1.